bundles/org.eclipse.equinox.common.tests/src/org/eclipse/equinox/common/tests/URIUtilTest.java

```diff
@@ -552,4 +561,74 @@ public void testMakeRelative() throws URISyntaxException {
 			assertEquals("2." + Integer.toString(i), expected, actual);
 		}
 	}
+
+	/**
+	 * Test UNC-Paths containing a $.
+	 */
+	public void testDollar() throws URISyntaxException {
+		var relative = "SomePath";
+		URI expectedResolved = new URI("file:////WSL$/Ubuntu/SomePath");
+		final String[] uris = {
+				// @formatter:off
+				"file:////WSL$/Ubuntu",
+				"file:////WSL$/Ubuntu/",
+				"file://////WSL$/Ubuntu",
+				"file://////WSL$/Ubuntu/"
+				// @formatter:on
+		};
+		final String[] expected_four = {
+				// @formatter:off
+				"file:////WSL$/Ubuntu",
+				"file:////WSL$/Ubuntu/",
+				"file:////WSL$/Ubuntu",
+				"file:////WSL$/Ubuntu/"
+				// @formatter:on
+		};
+
+		for (int i = 0; i < uris.length; i++) {
+			URI base = new URI(uris[i]);
+			URI resolved = URIUtil.append(base, relative);
+			assertEquals("1." + Integer.toString(i), expectedResolved, resolved);
```

Comment:
Can we please use **meaningfull** messages here? and of course also below?

Reply:
Of course. But I oriented myself by the testcases that were already there....

Reply:
Please don't do so, that are just "bad" examples that are hard to understand if something fails, so you better adjust these so you would understand failures without the code...

Reply:
Would you be happy with inline comments?  I think explaining things in the exception is rather longish. And one would have to look at the code anyway to understand what is going on.

Reply:
I don't think there is any length restriction in the message :-)
In this case I think something like "Appending "+relative+" to base URI "+base+" yields wrong results" (if I understand correctly what is going on here!)

Reply:
OK. I tried to make the exceptions more expressive and also extended the test-cases to show that the URI-ctors actually fail.